### PR TITLE
Nits/distribute

### PR DESCRIPTION
**Pull request by mkolesnik**
Fixing nits for PR #5

Rebase this after the PR is merged

### CODE REVIEW

File: pkg/federate/kubefed/resource_test.go [hunk 1]
```diff
@@ -227,33 +197,47 @@ func testDistribute() {
...
 				Expect(err).To(MatchError(HavePrefix("test error")))
 			})
 		})
 
 		When("the resource is not in the Kube API", func() {
 			It("should create the resource", func() {
 				// Sanity check
-				_, err := getResourceFromAPI(f.kubeFedClient, fedResource)
+				_, err = getResourceFromAPI(f.kubeFedClient, fedResource)
 				Expect(err).To(HaveOccurred())
```
Comment (tpantelis) [inline pkg/federate/kubefed/resource_test.go]: I think we should also verify the error indicates _NotFound_, ie

`Expect(apiErrors.IsNotFound(err)).To(BeTrue())`

File: pkg/federate/kubefed/resource_test.go [hunk 2]
```diff
@@ -50,95 +50,72 @@ func testCreateFederatedResource() {
...
-			})
+		It("should throw an error, failing to convert to Unstructured", func() {
+			var expectedType *UnstructuredConversionError
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(BeAssignableToTypeOf(expectedType))
+		})
+
+		It("should return a nil value", func() {
+			Expect(actual).To(BeNil())
 		})
```
[the remark this answers is not in the document]
Comment (mkolesnik) [reply pkg/federate/kubefed/resource_test.go]: Let's leave this for Michel to reason why he did it like this? :)

File: pkg/federate/kubefed/resource_test.go [hunk 3]
```diff
@@ -50,95 +50,72 @@ func testCreateFederatedResource() {
...
 	When("the resource type scheme has been added to the type registry", func() {
 		BeforeEach(func() {
-			err := corev1.AddToScheme(scheme)
+			err = corev1.AddToScheme(scheme)
 			Expect(err).ToNot(HaveOccurred())
 		})
 
-		It("should not modify the input resource but copy it", func() {
-			expected, _ := createFederatedResource(scheme, resource)
-			Expect(expected).ToNot(BeIdenticalTo(resource))
```
[the remark this answers is not in the document]
Comment (mkolesnik) [reply pkg/federate/kubefed/resource_test.go]: I'll leave that to Michel to implement if he likes, but comparing the federated resource to the original one (which was the original test) makes no sense (given that the federated resource is thoroughly checked in the latter tests anyway)

File: pkg/federate/kubefed/resource_test.go [hunk 4]
```diff
@@ -50,95 +50,72 @@ func testCreateFederatedResource() {
...
 			Expect(found).To(BeTrue(), "Hierarchy of the MatchLabels field missing!")
 			Expect(mapObj).To(BeEmpty())
 		})
-
-		It("the returned resource's namespace should be the input's namespace", func() {
-			expected, err := createFederatedResource(scheme, resource)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(expected.GetNamespace()).
-				To(Equal(resource.GetNamespace()))
-		})
```
[the remark this answers is not in the document]
Comment (mkolesnik) [reply pkg/federate/kubefed/resource_test.go]: It wasn't, I moved it to be closer to other tests (line 95) of it's kind (which was the nit I had for this on PR #5 )

File: pkg/federate/kubefed/resource_test.go (hunk 1 above)
Comment (mkolesnik) [reply pkg/federate/kubefed/resource_test.go]: You got it! :)

File: pkg/federate/kubefed/resource_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tpantelis) [reply pkg/federate/kubefed/resource_test.go]: yeah I agree the test doesn't add any value the way it is.

**Comment by mangelajo** [comment]
Based on current voting (2 approvals) merging. github is not counting aswin.

**Pull request merged**